### PR TITLE
Add Amazon EKS Resource Detector 

### PR DESCRIPTION
**Pull request by KKelvinLo**
## Which problem is this PR solving?

- Added EKS Resource Detector
- This is similar to the AWS EKS Resource Detector in [Java](https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk-extensions/aws-v1-support/src/main/java/io/opentelemetry/sdk/extension/trace/aws/resource/EksResource.java)

## Short description of the changes

- Added functionality of AWS EKS resource detector
- Added corresponding unit tests for AWS EKS detector

### CODE REVIEW

File: packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    private async _isEks(config: ResourceDetectionConfigWithLogger): Promise<boolean> {
+      const options = {
+        hostname: this.K8S_SVC_URL,
+        path: this.AUTH_CONFIGMAP_PATH,
+        method: 'GET',
+        timeout: this.MILLISECOND_TIME_OUT,
+        HEADERS: {
+          "Authorization" : this._getK8sCredHeader(config),
+        },
+        ca: JSON.stringify(AwsEksDetector.readFileAsync(this.K8S_CERT_PATH)),
```
Comment (anuraaga) [inline packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts]: Let's only read the CA file once

Comment (anuraaga) [reply packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts]: Also I think this is stringifying a promise, the unit test might be missing a sanity check of the content of this option

Comment (anuraaga) [reply packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts]: I suppose there shouldn't be a need to call JSON.stringify for the cert anyways

Comment (KKelvinLo) [reply packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts]: After looking into nock documentation, there doesn't seem to be a plausible way nock for a sanity check for the content of this option. Are there any alternatives?

**Review by anuraaga — APPROVED** [review]
Thanks!

**Review by alolita — COMMENTED** [review]
The naming conventions used are inconsistent.

The error messages need to be improved.

**Comment by KKelvinLo** [comment]
> The naming conventions used are inconsistent.
> 
> The error messages need to be improved.

Updated error messages and kubernetes(k8s calls to be more consistent).

**Pull request closed without merging**